packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -866,20 +933,47 @@ export function getPossibleTypes(
 		: [typeDef]
 }
 
+interface SimplifyNode<T> {
+	value: SimplifyValueNode<T> | undefined
+	/**
+	 * This key facilitates the %key lookup. This key should always be a member of
+	 * {@link value} and facilitates the `%key` lookup. This is the key of the member
+	 * the dispatcher originates from, while {@link value} is the struct containing
+	 * this key with a dispatcher associated with it.
+	 */
+	key: RuntimeNode<T> | undefined
+}
+/**
+ * Basically the same as a {@link CheckerTreeRuntimeNode}, just with a few fields
+ * that aren't needed here removed for simplicity.
```

Comment:
Not sure I'm a fan of introducing an entirely new set of types just because you don't need `children` here. That seems more confusing than reusing the `CheckerTreeRuntimeNode`. This rework makes the `simplify` function a bit more messy at first glance.

Reply:
The reason I do this is because it would be really annoying to need to construct a checker tree node with a bunch of empty arrays in e.g. line 1069, or require them to be set by another consumer.

Reply:
Ah, I missed that you also contruct these types from scratch. Makes sense to not reuse the types then.